File: java/src/main/java/section1/operator/Operator.java
package section1.operator;

public class Operator {

    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney money2 = money1;
        JavaMoney money3 = new JavaMoney(1_000L);

        System.out.println(money1 == money2); // true 동일성
        System.out.println(money1 == money3); // false 동등성

        if (money1.compareTo(money3) > 0) {
            System.out.println("money1 is greater than money2");
        }

        System.out.println(money1.plus(money2));
    }

}

File: kotlin/src/main/kotlin/operator/Operator.kt
package operator

fun main() {
    val money1 = JavaMoney(2_000L)
    val money2 = money1
    val money3 = JavaMoney(2_000L)

    println(money1 == money2) // true
    println(money1 == money3) // true
    println(money1 === money3) // false

    if (money1 > money3) {
        println("money1 is greater than money2")
    }

    val moneyK1 = Money(1_000L)
    val moneyK2 = Money(2_000L)
    println(moneyK1 + moneyK2)
}
